tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,389 @@
+    systemctl restart network.service;
+fi
+EOF
+        script_output($bridge_setup, 900);
+    }
+}
+
+sub restore_standalone {
+    #File standalone was installed from qa_test_virtualization package
+    my $standalone_path = "/usr/share/qa/qa_test_virtualization/shared/standalone";
+    assert_script_run("source $standalone_path", 60) if (script_run("[[ -f $standalone_path ]]") == 0);
+}
+
+sub destroy_standalone {
+    #File cleanup was installed from qa_test_virtualization package
+    my $cleanup_path = "/usr/share/qa/qa_test_virtualization/cleanup";
+    assert_script_run("source $cleanup_path", 60) if (script_run("[[ -f $cleanup_path ]]") == 0);
+}
+
+sub restart_libvirtd {
+    is_sle('>11') ? systemctl 'restart libvirtd' : script_run("service libvirtd restart");
+}
+
+sub restore_guests {
+    my $get_vm_hostnames   = "virsh list  --all | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames       = script_output($get_vm_hostnames, 30, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_array = split(/\n+/, $vm_hostnames);
+    foreach (@vm_hostnames_array)
+    {
+        script_run("virsh destroy $_");
+        script_run("virsh undefine $_");
+        script_run("virsh define $_.xml");
+        upload_logs "$_.xml";
+        script_run("rm -rf $_.xml");
+    }
+}
+
+sub restore_network {
+    my $network_mark = "/etc/sysconfig/network/ifcfg-br0.new";
+    if (script_run("[[ -f $network_mark ]]") == 0) {
+        assert_script_run("ip link set dev $virt_host_bridge down", 60);
+        my $bridge_destroy = <<'EOF';
+#!/bin/bash
+. /etc/os-release
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='dhcp'\nSTARTMODE='auto'"
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+rm -rf /etc/sysconfig/network/ifcfg-br0*
+if [[ $VERSION_ID =~ '11' ]]; then
+    service libvirtd restart;
+else
+    systemctl restart network.service;
+fi
+EOF
+        script_output($bridge_destroy, 900);
+    }
+}
+
+sub destroy_vir_network {
+    #Get the create virtual network name
+    my $get_vnet_name   = "virsh net-list --all| grep vnet | head -1 | awk \'{print \$1}\'";
+    my $vnet_name       = script_output($get_vnet_name, 30, type_command => 0, proceed_on_failure => 0);
+    my @vnet_name_array = split(/\n+/, $vnet_name);
+    foreach (@vnet_name_array) { script_run("virsh net-destroy $_"); script_run("rm -rf $_.xml"); }
+}
+
+sub restore_libvirt_default {
+    my $default_path = "/root/libvirt_default.xml";
+    if (script_run("[[ -f $default_path ]]") == 0) {
+        assert_script_run("virsh net-define $default_path", 60);
+        assert_script_run("rm -rf $default_path");
+    }
+}
+
+sub run {
+    #Prepare VM HOST SERVER Network Interface Configuration
+    #for libvirt virtual network testing
+    prepare_network;
+
+    #Install required packages
+    zypper_call '-t in iptables iputils bind-utils sshpass';
+
+    #Guest SSH Setup
```

Comment:
Where is the guest re-definition which discards the original guest network, for us related to br123, for qam br0? There is better to be such a function for this, as we talked before.

Reply:
For now, there was created func restore_network to take it as your mentioned. 
If it was not better, I can do more investigation about it.

Reply:
I think you misunderstood. This is for network, I mean guest definition without br123/br0.

Reply:
Understood, I will update this as your mentioned.

Reply:
This is a place that need new code.

Reply:
Got it

Reply:
Had finished add the step about redefine guest network interface before libvirt virtual network testing, please refer to the latest verification run link as below:
http://10.67.19.82/tests/265